### PR TITLE
Refactor popupWindow.less with Aphrodite

### LINKED ISSUE #7927: Refactor popupWindow.less with Aphrodite
## Test plan

https://github.com/brave/browser-laptop/pull/7929#issue-217486625

----

**Describe the issue you encountered:** Refactor popupWindow.less (used for extension dialogs) with Aphrodite

- Extra QA steps:
    1.
    2.
    3.

- Any related issues:

### PR DESCRIPTION
**Pull request by luixxiul**
Also add right:1em to create margin between the pop up and the window right border

Closes #7927

Auditors: @bbondy @bridiver @bsclifton @cezaraugusto @jonathansampson -  I would like to ask you to review the PR as you have edit [popupWindow.js](https://github.com/brave/browser-laptop/blob/2648803d46d3f02ce3c5dfb2f0e3007fec20642f/js/components/popupWindow.js) some way. Thanks in advance!

Test Plan:
1. Open about:preferences#security
2. Enable LastPass
3. Click the extension icon on the navigation bar
4. Make sure you can log in your account

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).

### CODE REVIEW

File: js/components/popupWindow.js [hunk 1]
```diff
@@ -69,35 +72,55 @@ class PopupWindow extends ImmutableComponent {
...
+        style.right = '1em'
       }
     }
 
     return <div
       className={cx({
         popupWindow: true,
-        reverseExpand: styles.right !== undefined
+        [css(styles.popupWindow)]: true,
+        [css((style.right !== undefined) && styles.reverseExpand)]: true
```
Comment (cezaraugusto) [inline js/components/popupWindow.js]: I know it was here before but not sure what's the reason of having it? row-reverse does nothing here

File: js/components/popupWindow.js [hunk 2]
```diff
@@ -69,35 +72,55 @@ class PopupWindow extends ImmutableComponent {
...
+    boxShadow: globalStyles.shadow.flyoutDialogBoxShadow,
+    boxSizing: 'border-box',
+    color: 'black',
+    cursor: 'default',
+    display: 'flex',
+    fontSize: '11px',
+    padding: 0,
+    position: 'absolute',
+    zIndex: globalStyles.zindex.zindexPopupWindow,
+    WebkitUserSelect: 'none'
```
Comment (jonathansampson) [inline js/components/popupWindow.js]: `user-select` (userSelect) is supported unprefixed in Blink. We should avoid using the prefixed version.

File: js/components/popupWindow.js [hunk 3]
```diff
@@ -69,35 +72,55 @@ class PopupWindow extends ImmutableComponent {
   }
 
   render () {
-    let styles = {}
+    let style = {}
     if (parseInt(this.width)) {
```
Comment (jonathansampson) [inline js/components/popupWindow.js]: Can we avoid doing `parseInt` twice for `this.width` and `this.height`?

File: app/renderer/components/styles/global.js [hunk 4]
```diff
@@ -129,6 +129,7 @@ const globalStyles = {
     switchNubShadow: '1px 1px 5px -2px black',
     buttonShadow: '0px 1px 5px -1px rgba(0, 0, 0, 1.0)',
     dialogShadow: '0px 8px 22px 0px rgba(0, 0, 0, .5)',
+    flyoutDialogBoxShadow: '2px 2px 8px #3b3b3b',
```
Comment (jonathansampson) [inline app/renderer/components/styles/global.js]: I don't think we should be using non-black hex colors for shadows. `rgba` handles this particular thing much better, IMHO.

File: js/components/popupWindow.js (hunk 3 above)
Comment (jonathansampson) [inline js/components/popupWindow.js]: Can we avoid doing `parseInt` twice for `this.width` and `this.height`?

File: js/components/popupWindow.js [hunk 5]
```diff
@@ -69,35 +72,55 @@ class PopupWindow extends ImmutableComponent {
...
-      styles.width = (parseInt(this.width) + 2)
+      style.width = (parseInt(this.width) + 2)
     }
     if (parseInt(this.height)) {
```
Comment (jonathansampson) [inline js/components/popupWindow.js]: Can we avoid doing `parseInt` twice for `this.width` and `this.height`?

File: js/components/popupWindow.js (hunk 2 above)
Comment (jonathansampson) [inline js/components/popupWindow.js]: `user-select` (userSelect) is supported unprefixed in Blink. We should avoid using the prefixed version.

File: less/variables.less [hunk 6]
```diff
@@ -117,6 +117,7 @@
 
 @buttonShadow: 0px 1px 5px -1px rgba(0, 0, 0, 0.5);
 @dialogShadow: 0px 8px 22px 0px rgba(0, 0, 0, .5);
+@flyoutDialogBoxShadow: 2px 2px 8px #3b3b3b;
```
Comment (jonathansampson) [inline less/variables.less]: To echo my earlier comment; I don't think we should use non-black HEX colors for shadows. `rgba` handles this much better.

File: js/components/popupWindow.js (hunk 1 above)
Comment (luixxiul) [inline js/components/popupWindow.js]: @bridiver do you remember? https://github.com/brave/browser-laptop/commit/4bf6154f3748224c2e51a9a2f1c269af929bec65#diff-19f896aac3ef3233d5725e9e002625ccR19

File: app/renderer/components/styles/global.js (hunk 4 above)
Comment (luixxiul) [inline app/renderer/components/styles/global.js]: Thanks for notifying. I'll replace it this with a variable later.

File: less/variables.less (hunk 6 above)
Comment (luixxiul) [inline less/variables.less]: ditto!

File: js/components/popupWindow.js (hunk 5 above)
Comment (bsclifton) [inline js/components/popupWindow.js]: ++ on this comment

**Comment by cezaraugusto** [comment]
rebase pls

**Comment by luixxiul** [comment]
Would anyone please add another commit to fix the issues mentioned above about parseInt? Thanks!

File: js/components/popupWindow.js [hunk 7]
```diff
@@ -72,22 +72,27 @@ class PopupWindow extends ImmutableComponent {
...
-    if (parseInt(this.height)) {
-      style.height = this.height + 2
+    if (height) {
+      style.height = height + 2
     }
-    if (parseInt(this.top)) {
-      if (this.top + this.height < window.innerHeight) {
-        style.top = this.top
+    if (top) {
+      if (top + this.height < window.innerHeight) {
```
Comment (jonathansampson) [inline js/components/popupWindow.js]: I think you meant `top + height` here.

**Comment by luixxiul** [comment]
thanks guys for helping me out :-)

**Pull request merged**